### DISCUSSION TITLE
Data Preperation script with support for date and time in different columns

### DISCUSSION THREAD

**Discussion by DanAE111** [patrickvorgers/Home-Assistant-Import-Energy-Data#37] [Ideas]
Hello,

I was looking at running my data through the preparation scripts so it could be imported into HA. The data from my provider is in CSV format with columns for date, time, consumption since the last read and some meter info. Somewhat similar to the existing Engie data source.

The issue i have is that the date and time are in separate columns while i was able to easily get around this with some concatenation in the source data i was wondering how trivial or difficult this would be to update the data preparation script for. Python is not one of my strengths unfortunately.

I've managed to setup the data import script in all ways except handling the separated date and time.

I believe many providers at least in my distribution area of Australia may use the same format. Here is an example.

AccountNumber | NMI | Register | ReadConsumption | ReadUnit | ReadQuality | ReadDate | ReadTime
-- | -- | -- | -- | -- | -- | -- | --
12345678 | 1234567890 | 1 | 0.679 | kWh | A | 2024/08/05 | 21:30:00
12345678 | 1234567890 | 1 | 1.358 | kWh | A | 2024/08/05 | 21:00:00
12345678 | 1234567890 | 1 | 0.891 | kWh | A | 2024/08/05 | 20:30:00
12345678 | 1234567890 | 1 | 0.718 | kWh | A | 2024/08/05 | 20:00:00
12345678 | 1234567890 | 1 | 0.757 | kWh | A | 2024/08/05 | 19:30:00
12345678 | 1234567890 | 1 | 0.585 | kWh | A | 2024/08/05 | 19:00:00
12345678 | 1234567890 | 1 | 0.591 | kWh | A | 2024/08/05 | 18:30:00

**Comment by patrickvorgers** [comment]
Good suggestion, I will have a look at it to see whether this can be supported.

**Comment by Stephan-4711** [comment]
If you need a quick and dirty way I used in a different scenario:
Open the file with excel, merge the Date and time in this way
in =CONCAT(G1,";",H1) should be the function in english excel. I was able to test with the german excel but there I have to use ; instead of .

After merging you can remove the formula, delete G and H and then export this back as csv :-|

**Comment by patrickvorgers** [comment]
@DanAE111 support has been added in enhancement #39
